src/Tizen.NUI.WindowSystem/src/public/InputGesture.cs

```diff
@@ -0,0 +1,557 @@
+                        Log.Debug(LogTag, "EdgeSwipe Event received. mode: " + mode + ", fingers: " + fingers);
+                        _edgeSwipeEventHandler?.Invoke(null, args);
+                    };
+                    Interop.InputGesture.ErrorCode res = Interop.InputGesture.SetEdgeSwipeCb(_handler, _edgeSwipeDelegate, IntPtr.Zero);
+                    ErrorCodeThrow(res);
+                }
+
+                _edgeSwipeEventHandler += value;
+            }
+            remove
+            {
+                _edgeSwipeEventHandler -= value;
+            }
+        }
+
+        /// <summary>
+        /// Emits the event when the edge drag event comes
+        /// </summary>
+        /// <exception cref="ArgumentException">Thrown when failed of invalid argument.</exception>
+        public event EventHandler<EdgeDragEventArgs> EdgeDragEventHandler
+        {
+            add
+            {
+                if (_edgeDragEventHandler == null)
+                {
+                    _edgeDragDelegate = (IntPtr userData, int mode,  int fingers, int cx, int cy, int edge) =>
+                    {
+                        EdgeDragEventArgs args = new EdgeDragEventArgs(mode, fingers, cx, cy, edge);
+                        Log.Debug(LogTag, "EdgeDrag Event received. mode: " + mode + ", fingers: " + fingers);
+                        _edgeDragEventHandler?.Invoke(null, args);
+                    };
+                    Interop.InputGesture.ErrorCode res = Interop.InputGesture.SetEdgeDragCb(_handler, _edgeDragDelegate, IntPtr.Zero);
+                    ErrorCodeThrow(res);
+                }
+
+                _edgeDragEventHandler += value;
+            }
+            remove
+            {
+                _edgeDragEventHandler -= value;
+            }
+        }
+
+        /// <summary>
+        /// Emits the event when the tap event comes
+        /// </summary>
+        /// <exception cref="ArgumentException">Thrown when failed of invalid argument.</exception>
+        public event EventHandler<TapEventArgs> TapEventHandler
+        {
+            add
+            {
+                if (_tapEventHandler == null)
+                {
+                    _tapDelegate = (IntPtr userData, int mode,  int fingers, int repeats) =>
+                    {
+                        TapEventArgs args = new TapEventArgs(mode, fingers, repeats);
+                        Log.Debug(LogTag, "Tap Event received. mode: " + mode + ", fingers: " + fingers + ", repeats: " + repeats);
+                        _tapEventHandler?.Invoke(null, args);
+                    };
+                    Interop.InputGesture.ErrorCode res = Interop.InputGesture.SetTapCb(_handler, _tapDelegate, IntPtr.Zero);
+                    ErrorCodeThrow(res);
+                }
+
+                _tapEventHandler += value;
+            }
+            remove
+            {
+                _tapEventHandler -= value;
```

Comment:
same as above

Reply:
Thank you for your comment. I added the code snippet unsetting callback function.